pypeit/core/gui/skysub_regions.py

```diff
@@ -0,0 +1,657 @@
+        matplotlib.pyplot.rcParams['keymap.save'] = ''              # saving current figure (Default: s)
+        matplotlib.pyplot.rcParams['keymap.quit'] = ''              # close the current figure (Default: ctrl+w, cmd+w)
+        matplotlib.pyplot.rcParams['keymap.grid'] = ''              # switching on/off a grid in current axes (Default: g)
+        matplotlib.pyplot.rcParams['keymap.yscale'] = ''            # toggle scaling of y-axes ('log'/'linear') (Default: l)
+        matplotlib.pyplot.rcParams['keymap.xscale'] = ''            # toggle scaling of x-axes ('log'/'linear') (Default: L, k)
+        matplotlib.pyplot.rcParams['keymap.all_axes'] = ''          # enable all axes (Default: a)
+
+        # Initialise the main canvas tools
+        canvas.mpl_connect('draw_event', self.draw_callback)
+        canvas.mpl_connect('button_press_event', self.button_press_callback)
+        canvas.mpl_connect('key_press_event', self.key_press_callback)
+        canvas.mpl_connect('button_release_event', self.button_release_callback)
+        canvas.mpl_connect('motion_notify_event',  self.mouse_move_callback)
+        self.canvas = canvas
+
+        # Interaction variables
+        self._respreq = [False, None]  # Does the user need to provide a response before any other operation will be permitted? Once the user responds, the second element of this array provides the action to be performed.
+        self._qconf = False  # Confirm quit message
+        self._changes = False
+        self._use_updates = True
+        self._inslit = -1  # Which slit is the mouse in
+        self.mmx, self.mmy = 0, 0
+        self._fitr = []  # Matplotlib shaded fit region
+        self._fita = None
+
+        # Draw the spectrum
+        self.canvas.draw()
+
+        self.initialize_menu()
+        self.reset_regions()
+
+    @classmethod
+    def initialize(cls, det, frame, tslits_dict, outname="skyregions.fits", runtime=False, printout=False):
+        """Initialize the 'ObjFindGUI' window for interactive object tracing
+
+            Args:
+                frame : ndarray
+                    Sky subtracted science image
+                tslits_dict : dict, None
+                    Dictionary containing slit trace information
+                det : int
+                    Detector index
+                printout : bool
+                    Should the results be printed to screen
+                runtime : bool
+                    Is this GUI being launched during a data reduction?
+
+            Returns:
+                ObjFindGUI: Returns an instance of the ObjFindGUI class
+        """
+        # This allows the input lord and rord to either be (nspec, nslit) arrays or a single
+        # vectors of size (nspec)
+        if tslits_dict['slit_left'].ndim == 2:
+            nslit = tslits_dict['slit_left'].shape[1]
+        else:
+            nslit = 1
+            tslits_dict['slit_left'] = tslits_dict['slit_left'].reshape((tslits_dict['slit_left'].size, 1))
```

Comment:
`tslits_dict` is on its way to Deprecation.  It will be replaced by the
`SlitTraceSet` object in the `rmtdict` branch.  I think it is ok to go
with this for now, but here is a head's up that we'll need to refactor
this soon.  Another knock-on effect of `demos_merge3` holding things up..

Reply:
Looks like you'll be able to handle it easily enough using `lordloc` and `rordloc`

Reply:
I think that's fine - I'm happy to refactor immediately after #817 passes.

Reply:
OK - I've merged `rmtdict` into branch `KCWI4` and refactored the skysub regions code to use the new `SlitTraceSet` - thanks for the heads up! I won't submit a PR for `KCWI4` until `rmtdict` is merged.